pkg/state/state_test.go

```diff
@@ -3477,3 +3477,99 @@ func TestHideChartURL(t *testing.T) {
 		}
 	}
 }
+
+func Test_appendExtraDiffFlags(t *testing.T) {
+	tests := []struct {
+		name          string
+		inputFlags    []string
+		inputOpts     *DiffOpts
+		inputDefaults []string
+
+		expected []string
+	}{
+		{
+			name:       "Skipping default flags, because diffOpts is provided",
+			inputFlags: []string{"aaaaa"},
+			inputOpts: &DiffOpts{
+				DiffArgs: "-bbbb --cccc",
+			},
+			inputDefaults: []string{"-dddd", "--eeee"},
+			expected:      []string{"aaaaa", "-bbbb", "--cccc"},
+		},
+		{
+			name:          "Use default flags, because diffOpts is not provided",
+			inputFlags:    []string{"aaaaa"},
+			inputDefaults: []string{"-dddd", "--eeee"},
+			expected:      []string{"aaaaa", "-dddd", "--eeee"},
+		},
+		{
+			name:          "Don't add non-flag arguments",
+			inputFlags:    []string{"aaaaa"},
+			inputDefaults: []string{"-d=ddd", "non-flag", "--eeee"},
+			expected:      []string{"aaaaa", "-d=ddd", "--eeee"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			result := (&HelmState{
+				ReleaseSetSpec: ReleaseSetSpec{
+					HelmDefaults: HelmSpec{
+						DiffArgs: test.inputDefaults,
```

Comment:
why diffargs?

Reply:
@yxxhero this is a test for the existing appendExtraDiffFlags function.
The test for the new appendExtraSyncFlags function is at the bottom of this file.

Reply:
I added a test for the existing function since there was no test yet & both functions are very similar.

Reply:
@inteon oh. sorry. got it.